VocaDbWeb.Core/Controllers/SongController.cs

```diff
@@ -0,0 +1,590 @@
+			{
+				return HttpStatusCodeResult(HttpStatusCode.BadRequest, "File too large");
+			}
+
+			var pv = new LocalFileManager().CreatePVContract(new AspNetCoreHttpPostedFile(file), User.Identity, PermissionContext.LoggedUser);
+
+			return LowercaseJson(pv);
+		}
+
+		[ResponseCache(Location = ResponseCacheLocation.Client, Duration = 3600)]
+		public ActionResult PopupContent(int id = InvalidId)
+		{
+			if (id == InvalidId)
+				return NotFound();
+
+			var song = _queries.GetSong(id);
+			return PartialView("SongPopupContent", song);
+		}
+
+		[ResponseCache(Location = ResponseCacheLocation.Client, Duration = 3600)]
+		public async Task<ActionResult> PopupContentWithVote(int id = InvalidId, int? version = null, string callback = null)
+		{
+			if (id == InvalidId)
+				return NotFound();
+
+			var song = _queries.GetSongWithPVAndVote(id, false, includePVs: false);
+
+			if (string.IsNullOrEmpty(callback))
+			{
+				return PartialView("_SongWithVotePopupContent", song);
+			}
+			else
+			{
+				return Json(await RenderPartialViewToStringAsync("_SongWithVotePopupContent", song), callback);
+			}
+		}
+
+		// TODO: implement
+		/*public FeedResult Feed(IndexRouteParams indexParams)
+		{
+			WebHelper.VerifyUserAgent(Request);
+
+			var pageSize = (indexParams.pageSize.HasValue ? Math.Min(indexParams.pageSize.Value, 30) : 30);
+			var sortRule = indexParams.sort ?? SongSortRule.Name;
+			var timeFilter = DateTimeUtils.ParseFromSimpleString(indexParams.since);
+			var filter = indexParams.filter;
+			var songType = indexParams.songType ?? SongType.Unspecified;
+			var matchMode = indexParams.matchMode ?? NameMatchMode.Auto;
+			var onlyWithPVs = indexParams.onlyWithPVs ?? false;
+			var minScore = indexParams.minScore ?? 0;
+
+			var textQuery = SearchTextQuery.Create(filter, matchMode);
+			var queryParams = new SongQueryParams(textQuery,
+				songType != SongType.Unspecified ? new[] { songType } : new SongType[] { },
+				0, pageSize, false, sortRule, false, false, null)
+			{
+				ArtistParticipation = {
+						ArtistIds = !indexParams.artistId.HasValue || indexParams.artistId.Value == 0 ? null : new [] { indexParams.artistId.Value }
+					},
+				TimeFilter = timeFilter,
+				OnlyWithPVs = onlyWithPVs,
+				MinScore = minScore,
+				UserCollectionId = indexParams.userCollectionId ?? 0,
+				FollowedByUserId = indexParams.followedByUserId ?? 0
+			};
+
+			var result = Service.FindWithThumbPreferNotNico(queryParams);
+
+			var fac = new SongFeedFactory();
+			var feed = fac.Create(result.Items,
+				VocaUriBuilder.CreateAbsolute(Url.Action("Index", indexParams)),
+				song => RenderPartialViewToString("SongItem", song),
+				song => Url.Action("Details", new { id = song.Id }));
+
+			return new FeedResult(new Atom10FeedFormatter(feed));
+		}*/
+
+		// TODO: implement
+		/*public FeedResult LatestVideos()
+		{
+			return Feed(new IndexRouteParams { onlyWithPVs = true, pageSize = 20, sort = SongSortRule.AdditionDate });
+		}*/

```

Comment:
Needs to be implemented. Perhaps related to #713.

Reply:
These are not related to #713. These are RSS feeds that are being used by some people to follow new songs added.